frontend/src/pages/login/registerPage.tsx

```diff
@@ -0,0 +1,210 @@
+  const [showTermsModal, setShowTermsModal] = useState(false)
+
+  const [registerLoading, setRegisterLoading] = useState(false)
+
+  const [presentToast] = useIonToast()
+
+  const registerContext = useRegister()
+
+  const customPopoverOptions = {
+    header: 'Institution auswählen:',
+    cssClass: 'institution-select',
+  }
+
+  return (
+    <IonPage>
+      <IonHeader>
+        <IonToolbar>
+          <IonButtons slot="start">
+            <IonBackButton defaultHref="/" />
+          </IonButtons>
+          <IonTitle slot="primary">Metapholio</IonTitle>
+        </IonToolbar>
+      </IonHeader>
+      <IonContent>
+        <IonHeader collapse="condense">
+          <IonToolbar>
+            <IonTitle size="large">Metapholio</IonTitle>
+          </IonToolbar>
+        </IonHeader>
+        <div className="container">
+          <h2>Name:</h2>
+          <IonCard class="name-card">
+            <IonCardContent>
+              <IonItemGroup>
+                <IonItem class="ion-no-padding">
+                  <IonLabel position="stacked">Vorname</IonLabel>
+                  <IonInput
+                    autocomplete="name"
+                    inputmode="text"
+                    pattern="text"
+                    placeholder="Vorname eingeben"
+                    type="text"
+                    value={forename}
+                    onIonChange={e => setForename(e.detail.value ?? '')}
+                  ></IonInput>
+                </IonItem>
+                <IonItem class="ion-no-padding">
+                  <IonLabel position="stacked">Nachname</IonLabel>
+                  <IonInput
+                    autocomplete="family-name"
+                    inputmode="text"
+                    pattern="text"
+                    placeholder="Nachname eingeben"
+                    type="text"
+                    value={surname}
+                    onIonChange={e => setSurname(e.detail.value ?? '')}
+                  ></IonInput>
+                </IonItem>
+              </IonItemGroup>
+            </IonCardContent>
+          </IonCard>
+
+          <h2>Institution:</h2>
+          <IonCard className="name-card">
+            <IonCardContent>
+              <IonItem class="ion-no-padding">
+                <IonSelect
+                  interfaceOptions={customPopoverOptions}
+                  className="institution-select"
+                  value={institution}
+                  onIonChange={e => setInstitution(e.detail.value)}
+                  interface="action-sheet"
+                  placeholder="Institution auswählen"
+                  cancelText="Abbrechen"
+                  okText="Auswählen"
+                >
+                  {institutions.map((e, index) => (
+                    <IonSelectOption
+                      className="institution-select"
+                      key={index}
+                      value={e}
+                    >
+                      {e}
+                    </IonSelectOption>
```

Comment:
e?
```suggestion
                  {institutions.map((institution, index) => (
                    <IonSelectOption
                      className="institution-select"
                      key={`institution-selection-${index}`}
                      value={institution}
                    >
                      {institution}
                    </IonSelectOption>
```

index is only a countable value starting from 0, when you use only index it can be possible that there is another component with the same key.

Reply:
Makes sense. In normal situations where I have full control of the markup that shouldn't be a problem as it is scoped to this `map` function. 
But not sure what's happening under the hood.